### ISSUE TITLE
Question: set signal sent from pm2 when restart triggered by `max_memory_restart`.

### ISSUE THREAD

**Issue by mishawakerman** [Unitech/pm2#4077]
// Copying [question from Slack](https://pm2-nodejs.slack.com/archives/C1PV6Q5RP/p1544740997013000)

**tl;dr is there a way to differentiate the signals sent to the process in the event of the `max_memory_restart` threshold being exceeded and the event of being restarted by `startOrGracefulReload`?**

We use `pm2` to manage some job processes that are continuously deployed via `pm2 startOrGracefulReload`. These processes have internal handlers that catch the `SIGINT` restart signal  and set the running job back to `progress=0` and `status=PENDING` for the newly deployed job to pickup when it starts up. This all works flawlessly, thanks `pm2`!

We recently found that under conditions (that we can't check ahead of time), a job may be too large and cause the server to run out of memory. In this circumstance, we are happy to abort the job and mark it as failed. Using `max_memory_restart` would be a great way to do this, but it sends the same `SIGINT` signal to the process as when we reload for deployment purposes. This means we can't differentiate a reload caused by memory thresholds being exceeded and those caused by our deployment pipeline. And the handlers for these two situations need to do very different things.

So, we'd like to be able to define the signal sent in the event of `max_memory_threshold` being exceeded by a process or at least differentiate the events above. Is this currently possible with `pm2`?
